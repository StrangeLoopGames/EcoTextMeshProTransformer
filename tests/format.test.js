const { transform, Tag } = require('../dist/index.js');

const testCallback = (openTag, content) => {
  switch (openTag.tag) {
    case Tag.Color: {
      return `[myColor=${openTag.value}]${content}[/myColor]`;
    }
    case Tag.Underline:
      return `[underline]${content}[/underline]`;
    case Tag.Italic:
      return `[italic]${content}[/italic]`;
    case Tag.Bold:
      return `[bold]${content}[/bold]`;
  }

  return content;
};

describe('transform', function () {
  test('simple', function () {
    const result = transform(`
    was <color=red>test <color=blue><u><b>wd</b></u></color> xad
    `, testCallback);

    expect(result).toMatchSnapshot();
  });

  test('without close tags', function () {
    const result = transform(`
    was <color=red>test <color=blue><u><b>wd xad
    `, testCallback);

    expect(result).toMatchSnapshot();
  });

  test('unknown tags', function () {
    const result = transform(`
    was <color=red>test <unknown>wd xad
    `, testCallback);

    expect(result).toMatchSnapshot();
  });

  test('invalid closing of tag', function () {
    const result = transform(`
    was <color=red>test <b>wd xad</color>
    `, testCallback);

    expect(result).toMatchSnapshot();
  });

  test('two top level tags', function () {
    const result = transform(`
    was <color=red>test</color> <color=blue><b>wd xad
    `, testCallback);

    expect(result).toMatchSnapshot();
  });

  test('two top level tags with inner styles', function () {
    const result = transform(`
    was <color=red>te<u>st</u></color> <color=blue><b>wd xad
    `, testCallback);

    expect(result).toMatchSnapshot();
  });

  test('style without value', function () {
    const result = transform(`
    was <link>test</link>
    `, testCallback);

    expect(result).toMatchSnapshot();
  });
});
