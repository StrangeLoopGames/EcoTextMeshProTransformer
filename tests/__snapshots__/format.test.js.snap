// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform invalid closing of tag 1`] = `
"
    was [myColor=red]test [bold]wd xad</color>
    [/bold][/myColor]"
`;

exports[`transform multiline 1`] = `
"
    [myColor=red]
    title - text

    -a --abc - [bold]description[/bold]
    -a --abc - [bold]description[/bold]
    -a --abc - [bold]description[/bold]
    -a --abc - [bold]description[/bold]
    [/myColor]
    "
`;

exports[`transform simple 1`] = `
"
    was [myColor=red]test [myColor=blue][underline][bold]wd[/bold][/underline][/myColor] xad
    [/myColor]"
`;

exports[`transform style without value 1`] = `
"
    was test
    "
`;

exports[`transform two top level tags 1`] = `
"
    was [myColor=red]test[/myColor] [myColor=blue][bold]wd xad
    [/bold][/myColor]"
`;

exports[`transform two top level tags with inner styles 1`] = `
"
    was [myColor=red]te[underline]st[/underline][/myColor] [myColor=blue][bold]wd xad
    [/bold][/myColor]"
`;

exports[`transform unknown tags 1`] = `
"
    was [myColor=red]test <unknown>wd xad
    [/myColor]"
`;

exports[`transform without close tags 1`] = `
"
    was [myColor=red]test [myColor=blue][underline][bold]wd xad
    [/bold][/underline][/myColor][/myColor]"
`;
